pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -77,6 +76,40 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	if subnetPort.ObjectMeta.DeletionTimestamp.IsZero() {
+		if len(subnetPort.Spec.Subnet) != 0 {
+			s := &v1alpha1.Subnet{}
+			sNamespacedName := types.NamespacedName{
+				Namespace: subnetPort.Namespace,
+				Name:      subnetPort.Spec.Subnet,
+			}
+			if err := r.Client.Get(ctx, sNamespacedName, s); err != nil {
+				log.Error(err, "unable to fetch subnet CR", "req", req.NamespacedName)
+				return common.ResultRequeueAfter10sec, err
+			}
+			if s != nil && !s.DeletionTimestamp.IsZero() {
+				err := fmt.Errorf("subnet %s is been deleting, cannot operate subnetport %s", s.Name, req.NamespacedName)
+				updateFail(r, ctx, subnetPort, &err)
+				return common.ResultNormal, err
+			}
+		}
+
+		if len(subnetPort.Spec.SubnetSet) != 0 {
+			subnets := &v1alpha1.SubnetSet{}
+			sNamespacedName := types.NamespacedName{
+				Namespace: subnetPort.Namespace,
+				Name:      subnetPort.Spec.SubnetSet,
+			}
+			if err := r.Client.Get(ctx, sNamespacedName, subnets); err != nil {
+				log.Error(err, "unable to fetch SubnetSet CR", "req", req.NamespacedName)
+				return common.ResultRequeueAfter10sec, err
+			}
+			if subnets != nil && !subnets.DeletionTimestamp.IsZero() {
+				err := fmt.Errorf("SubnetSet %s is been deleting, cannot operate subnetport %s", subnets.Name, req.NamespacedName)
```

Comment:
ditto. `SubnetSet %s is been deleting` -> `subnetset %s is being deleted`

Reply:
done